pretyping/pretyping.ml

```diff
@@ -207,34 +207,47 @@ type pretype_flags = {
    [sigma'] into those already in [sigma] or deriving from an evar in
    [sigma] by restriction, and the evars properly created in [sigma'] *)
 
-type frozen =
-| FrozenId of undefined evar_info Evar.Map.t
-  (** No pending evars. We do not put a set here not to reallocate like crazy,
-      but the actual data of the map is not used, only keys matter. All
-      functions operating on this type must have the same behaviour on
-      [FrozenId map] and [FrozenProgress (Evar.Map.domain map, Evar.Set.empty)] *)
-| FrozenProgress of (Evar.Set.t * Evar.Set.t) Lazy.t
-  (** Proper partition of the evar map as described above. *)
+type frozen_and_pending =
+  Frz :
```

Comment:
I think that the FrozenId case should be preserved even if the code path is similar, since it makes explicit the trivial invariant.

Reply:
I don't see the point

Reply:
Explicit is better than implicit, especially when dealing with pointer equality.

Reply:
How is FrozenId more explicit than Evar.Set.empty?

Reply:
No arbitrary lazy code execution amongst others.

Reply:
Is that really worth  having more code to have?

Reply:
It's less diff lines and it's a trivial case.

Reply:
cf new commit if you insist

Reply:
I was hoping that you would keep the two previous constructors...

Reply:
please stop nitpicking thi spr